### DISCUSSION TITLE
Write TIFF metadata and preserve it through maketx conversion

### DISCUSSION THREAD

**Discussion by vincentgires** [OpenImageIO/oiio#4343] [General]
Hello,
We're using oiiotool/maketx to convert images to 8 bits tiff/tx and we would add some custom metadata. Does the --attrib/--sattrib arguments of oiiotool are supposed to work with TIFF files? this doesn't add metadata (checked with iinfo, blender and nuke to be sure) but it's adding metadata correctly with the same command on EXR file format. Is that a limitation of the TIFF format or this is only not implemented in OpenImageIO?

Tested with 2.5.12.0.
Thanks.

**Comment by lgritz** [comment]
What metadata do you want to add?

--attrib "works with" TIFF files, but you have to remember that TIFF only can store metadata that corresponds to the registered tags it understands in the header. This is different from, say, OpenEXR, which allows arbitrary name/value metadata to be added without any prior registration of those names in the format specification.

**Comment by lgritz** [comment]
Phrased another way: there are hundreds of metadata items you can put in a TIFF file, but it's a fixed list. OpenEXR lets you put metadata with any name you like into the file.

**Comment by vincentgires** [comment]
Thank you for your answer. Indeed, I tried with custom tag name and thought TIFF would allow free names like EXR and MKV do. I will take a look to available metadata names and find which one can match to our needs. Thanks!
